### PR TITLE
Datasets quick fixes + SQLAlchemy addition + CFF updates

### LINKED ISSUE #316: (Python) database tools / chapter?
The Python chapter contains a section "Database Interface" that doesn't seem helpful. As far as I know, what we use from Python to access databases is typically SQLAlchemy, [as @bouweandela suggested](https://github.com/NLeSC/guide/pull/302/files#r1383205726). I have used this personally in the past, but it's been a long time and I don't feel confident enough on this topic to write more than just "SQLAlchemy good".

Does anyone with more recent experience want to write something about this? @HannoSpreeuw maybe? Needn't be more than a single sentence, but if you have good tips ("ALWAYS rtfm first" or "NEVER rtfm, it's horrible, try this instead" or...), that would be great as well.

Also, if anyone disagrees with the above assessment (that the currently listed database interface tools are essentially useless to our typical usecases), please correct me!

### PR DESCRIPTION
**Pull request by egpbos**
# Changes in this PR

Fixes:
- Layout fixes in the datasets chapter that was added today (I **did** do the local preview this time that the checklist told me to do 😅).
- Added @suvayu, @f-hafner and @recap as page owners, if you all agree :)
- Added the authors of the datasets chapter to the CFF file (...again, as the checklist suggested 🫣)

Adds:
- In #316 we discussed (among many things) replacing all the Python chapter database interfaces with SQLalchemy. I now added mention of this to the databases chapter. Closes #316.
- The CFF file was not updated much for the past 5 years. I went through past closed PRs to add the author names. If people want to add their ORCID, feel free to suggest in a review comment, or simply push it to this branch.

# Checklist

- [x] I previewed my changes locally using e.g. `python3 -m http.server 4000` and confirmed they work correctly.
- [x] My name **AND MANY OTHERS ;)** was added to the `CITATION.cff` file.

### CODE REVIEW

File: best_practices/datasets.md [hunk 1]
```diff
@@ -40,14 +42,23 @@ SQLite is a transactional database, so if you have a dataset that is changing wi
    - For both DuckDB and SQLite, unique indexes allow to ensure data integrity
    - For SQLite, indexes are crucial to improve the performance of queries. However, having more indexes makes writing new records to the database slower. So it's again a trade-off between query and write speed.
 
+# Useful libraries
+
+## Database APIs
+
+- [SQLAlchemy](https://www.sqlalchemy.org/)
+   - In Python, interfacing to SQL databases like SQLite, MySQL or PostgreSQL is often done using [SQLAlchemy](https://www.sqlalchemy.org/), which is an Object Relational Mapper (ORM) that allows you to map tables to Python classes. Note that you still need to use a lot of manual SQL outside of Python to manage the database. However, SQLAlchemy allows you to use the data in a Pythonic way once you have the database layout figured out.
+
```
Comment (suvayu) [inline best_practices/datasets.md:51]: I was thinking maybe [piccolo](https://github.com/piccolo-orm/piccolo) deserves a mention here, considering it's much simpler, and probably better fit for smaller projects.

**Review by suvayu — APPROVED** [review]

File: best_practices/datasets.md (hunk 1 above)
Comment (egpbos) [reply best_practices/datasets.md:51]: Please, by all means, add it. I am not at all an expert on databases, this is just a thing I happen to have used once.

Comment (egpbos) [reply best_practices/datasets.md:51]: Or perhaps we can do that in another PR and I'll just make an issue about it first to remember (maybe others have additional ideas that they can discuss there).

Comment (egpbos) [reply best_practices/datasets.md:51]: Done #346. Thanks for the review!

**Pull request merged**